### DISCUSSION TITLE
no full_rename?

### DISCUSSION THREAD

**Discussion by mosheavni** [nvim-tree/nvim-tree.lua#2145] [General]
I migrated to the `on_attach` function, but I cannot seem to find `full_rename` as it was before!
I tried all of those:

```lua
  vim.keymap.set('n', 'r', api.fs.rename, opts 'Rename')
  vim.keymap.set('n', 'r', api.fs.rename_node, opts 'Rename node')
  vim.keymap.set('n', 'r', api.fs.rename_basename, opts 'Rename basename')
  vim.keymap.set('n', 'r', api.fs.rename_sub, opts 'Rename sub')
```
(of course each time a different line from here)
nothing gives me both the full dirname and the filename, it's always one of them.
Is this feature gone?

**Comment by alex-courtis** [comment]
`full_rename` has been migrated to `api.fs.rename_sub`

`full_rename` was changed at 949913f to rename sub.

Options:
1. Add `api.fs.rename_full`
2. Allow rename to take a user argument e.g. `:p`

1 is consistent.
2 is not be achievable as specific handling of each modifier is required

A generic rename with `opts.mode` could be added however that adds little.

**Reply by mosheavni** [reply]
> `full_rename` has been migrated to `api.fs.rename_sub`
> 
> `full_rename` was changed at [949913f](https://github.com/nvim-tree/nvim-tree.lua/commit/949913f1860eb85024fa1967dbd89ac797777b0d) to rename sub.
> 
> Options:
> 
> 1. Add `api.fs.rename_full`
> 2. Allow rename to take a user argument e.g. `:p`
> 
> 1 is consistent. 2 is not be achievable as specific handling of each modifier is required
> 
> A generic rename with `opts.mode` could be added however that adds little.

so in short, this does not exist?
what can I do? this is one of the things I used the most in nvim-tree.lua 🥲

**Reply by alex-courtis** [reply]
Tracked here: #2148

You're not the only one.
